src/components/user/services/userProfile.service.js

```diff
@@ -70,19 +70,34 @@ angular.module('znk.infra.user').service('UserProfileService',
         }
 
         function _setProfile(newProfile, userId) {
+            var saveProfileProm = [];
             var authData = AuthService.getAuth();
             if (authData || userId){
                 var uid = userId ? userId : authData.uid;
                 var profilePath = 'users/' + uid + '/profile';
                 return UserStorageService.get(profilePath).then(function (profile) {
-                    return profile ? UserStorageService.update(profilePath, newProfile) : UserStorageService.set(profilePath, newProfile);
+                    if (ENV.setUserProfileTwice) {
+                        saveProfileProm.push(_setUserProfileTwice(profilePath, newProfile));
+                    }
+                    if (profile){
+                        saveProfileProm.push(UserStorageService.update(profilePath, newProfile));
+                    } else {
+                        saveProfileProm.push(UserStorageService.set(profilePath, newProfile));
+                    }
+                    return $q.all(saveProfileProm);
```

Comment:
who uses the return value ? in the previous code, you returned the profile....here you'll return an array